### DISCUSSION TITLE
Highlighting elements when dragging

### DISCUSSION THREAD

**Discussion by HMilbradt** [fabricjs/fabric.js#9825] [Q&A] [answered]
Hey there 👋 

I've been using Fabric for quite some time, and have a use case that I can't seem to find any information on.

Basically what I'm after is a way to highlight objects when using the multi-selection rectangle, as a sort of preview into what will be selected when the user lifts the mouse button.

For example, this screenshot shows a "hover" style I've added when I mouse over an object.  This is just manually showing the border of the object without controls on mouse over, and hiding them on mouse leave.
![image](https://github.com/fabricjs/fabric.js/assets/22685915/e7aec9ef-9c44-43ab-b42b-f4cd9348095e)

The next screenshot shows me in the "rectangular selection" mode.  
![image](https://github.com/fabricjs/fabric.js/assets/22685915/c644a2d4-8ce2-4167-aa5f-0d69dab657a6)

Ideally what I'd like to do is show the hover styles that I've applied in the first screenshot when the selection area intersects with the object somewhere.

Does anyone know if Fabric has something like this built into the codebase?  I have a prototype working that's fairly complicated, as it's looping over all objects and manually determining the intersection, but I'm curious if anyone has come across a "built-in" way to achieve this.

Thanks!

**Comment by HMilbradt** [comment]
For what it's worth, I ended up with something like this:

```
const pointer = canvas.getPointer(e)

const width = Math.abs(pointer.x - initialDragPosition.x)
const height = Math.abs(pointer.y - initialDragPosition.y)

const top =
  pointer.y > initialDragPosition.y ? initialDragPosition.y : pointer.y
const left =
  pointer.x > initialDragPosition.x ? initialDragPosition.x : pointer.x

const objects = app.getElements()

const tl = new fabric.Point(left, top)
const br = new fabric.Point(left + width, top + height)

objects.forEach((object) => {
  if (
    object.isContainedWithinRect(tl, br, true) ||
    object.intersectsWithRect(tl, br, true)
  ) {
    showHoverStyles(object)
  }
})
```

This works perfectly for my purposes, but I'm not sure if it's the "best" way.

**Comment by asturur** [comment]
I think you should be able to use the same code that will end up selecting the object themselves. Let me dig it out for you

**Reply by asturur** [reply, marked as answer]
You can use `Canvas.collectObjects` to get all the objects in the rectangle.
If you are using the onSelect selection control then you should also filter them and check which one is selectable.
But the 'collectObjects' method should be the method that represent the code you wrote

**Reply by HMilbradt** [reply]
Thanks @asturur, I'll take a look and will report back if there's an issue, but otherwise marking as the answer
